server/master/src/main/java/org/apache/accumulo/master/tableOps/tableImport/MapImportFileNames.java

```diff
@@ -51,66 +51,69 @@
+          String[] sa = fileName.split("\\.");
+          String extension = "";
+          if (sa.length > 1) {
+            extension = sa[sa.length - 1];
+
+            if (!FileOperations.getValidExtensions().contains(extension)) {
+              continue;
+            }
+          } else {
+            // assume it is a map file
+            extension = Constants.MAPFILE_EXTENSION;
+          }
 
-      for (FileStatus fileStatus : files) {
-        String fileName = fileStatus.getPath().getName();
-        log.info("filename " + fileStatus.getPath());
-        String[] sa = fileName.split("\\.");
-        String extension = "";
-        if (sa.length > 1) {
-          extension = sa[sa.length - 1];
+          String newName = "I" + namer.getNextName() + "." + extension;
 
-          if (!FileOperations.getValidExtensions().contains(extension)) {
-            continue;
-          }
-        } else {
-          // assume it is a map file
-          extension = Constants.MAPFILE_EXTENSION;
+          mappingsWriter.append(fileName);
+          mappingsWriter.append(':');
+          mappingsWriter.append(newName);
+          mappingsWriter.newLine();
         }
 
-        String newName = "I" + namer.getNextName() + "." + extension;
-
-        mappingsWriter.append(fileName);
-        mappingsWriter.append(':');
-        mappingsWriter.append(newName);
-        mappingsWriter.newLine();
+        mappingsWriter.close();
+        mappingsWriter = null;
+      } catch (IOException ioe) {
+        log.warn("{}", ioe.getMessage(), ioe);
+        throw new AcceptableThriftTableOperationException(tableInfo.tableId.canonical(),
+            tableInfo.tableName, TableOperation.IMPORT, TableOperationExceptionType.OTHER,
+            "Error writing mapping file " + path + " " + ioe.getMessage());
+      } finally {
+        if (mappingsWriter != null)
+          try {
+            mappingsWriter.close();
+          } catch (IOException ioe) {
+            log.warn("Failed to close " + path, ioe);
+          }
       }
-
-      mappingsWriter.close();
-      mappingsWriter = null;
-
-      return new PopulateMetadataTable(tableInfo);
-    } catch (IOException ioe) {
-      log.warn("{}", ioe.getMessage(), ioe);
-      throw new AcceptableThriftTableOperationException(tableInfo.tableId.canonical(),
-          tableInfo.tableName, TableOperation.IMPORT, TableOperationExceptionType.OTHER,
-          "Error writing mapping file " + path + " " + ioe.getMessage());
-    } finally {
-      if (mappingsWriter != null)
-        try {
-          mappingsWriter.close();
-        } catch (IOException ioe) {
-          log.warn("Failed to close " + path, ioe);
-        }
     }
+
+    return new PopulateMetadataTable(tableInfo);
   }
 
   @Override
   public void undo(long tid, Master env) throws Exception {
-    env.getFileSystem().deleteRecursively(new Path(tableInfo.importDir));
+    // TODO: will this be OK for partially complete operations?
```

Comment:
I would think if anything exists in the directory, then deleting it seems very dangerous....

Reply:
These directories were created in the `call` method and will be empty if the `undo` steps from future stages of the fate transaction properly `undo` their work. This should be safe as a result. I can likely remove this comment.